src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -157,30 +170,33 @@ private async ValueTask HighlightExpandAndPresentItemsAsync(ImmutableSegmentedLi
             if (model is null)
                 return;
 
-            // Switch to the UI thread to get the current caret point and latest active text view then create the UI model.
+            // Switch to the UI thread to get the current caret point and create the UI model.
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            var activeTextView = GetLastActiveIWpfTextView();
-            if (activeTextView is null)
-                return;
+            var caretPoint = GetActiveTextViewCaretPoint();
 
-            var caretPoint = activeTextView.GetCaretPoint(activeTextView.TextBuffer);
-            if (!caretPoint.HasValue)
-                return;
+            SnapshotPoint? GetActiveTextViewCaretPoint()
+            {
+                var activeTextView = GetLastActiveIWpfTextView();
+                if (activeTextView is null)
+                    return null;
+
+                return activeTextView.GetCaretPoint(activeTextView.TextBuffer);
+            }
```

Comment:
I updated this code since if we fail to get the active text view or the caret point, I don't think we should bail out - I think we should just not highlight any symbol node but still update the UI with the latest data.

Reply:
i like that idea a lot.  i think you shoudl apply that philosophy to the rest of hte processing you do.